### ISSUE TITLE
IfcPolyline / IfcPolynomialCurve for guidelines?

### ISSUE THREAD

**Issue by ivaroppen** [bSI-InfraRoom/IFC-Tunnel-Deployment#167]
I have tested several curves for parametric description of the guideline parameter along the peg number axis.

It has been found a 5-order polynomial curve provide superb accuracy (better than 1 micro meter in my test case: A T9.5 to T12.5 over 10m).

I need some advice on how to output the curve to the ifc file:

Currently I am creating a IfcPolyline from the control points.
In this manner the user of the file can fit a curve of own choice, or use a simple linear interpolation on the control points….

An alternative would be to create a IfcPolynomialCurve?

**Comment by SergejMuhic** [comment]
@ivaroppen I am not sure I understand what the scenario is. I also need some more information on what you are trying to achieve. Could you provide a little more context?

I am also not sure if you are using "control points" in the notion of a IfcBSplineCurve or if these have some other meaning because in that case you might want to look at bspline definitions.

An IfcPolynomialCurve is a parametric representation of a polynomial in a particular axis placement so yes, a 5th-order polynomial is definitely possible. It just does not come with explicit points like IfcPolyline.

First thing is first though, a bit more context would be very welcome.

**Comment by ivaroppen** [comment]
The control points in this context are the correct values for a set of peg numbers. For each parameter i try to fit a paramteric line. Paramters are: side offset, heigth offset and radius.

The first try was to fit a Bezier spline. This was abandoned because the Bezier, by nature, does not go through the control points. 

As a result, I tried a polynomial curve. Unfortunately, I kept the word “control points” in my description. I use standard maths to fit the polynomial to these values. For a 5-order polynomial I achieved excellent accuracy.

The question is:
Should I put the coefficient of the polynomial into the ifc file, ie. IfcPolynomialCurve, or should I put the peg numbers and calculated correct values (control points) into the ifc file as IfcPolyline.

For ifc IfcPolyline the producer of the file does not have to bother about curve fitting. The only assumption is to output enough value/peg number pairs. From my experiments 5-6 pairs for each parameter should be enough…
For the IfcPolynomialCurve the producer of the file has more control of the accuracy achieved by the consumer, but the cost is a bit higher complexity ie. curve fitting is needed… 

![BeverControl_Detail](https://github.com/bSI-InfraRoom/IFC-Tunnel-Deployment/assets/123669643/7fed1713-0b28-4edf-87b6-970fa707e165)
The cyan lines are the guide lines. Contol points are producer calculated values along these lines. The producer is assumed to calulate these values with high accuracy. Each cyan line will produce 3 lines in the curveset of the ifc file: side offset, height offset and radius.

----
Another question: I see the guided sweep has been merged into the schema, and the schema is built successfully. However, I still can not find it in the documentation at the link in your FAQ. Is there a separate build for the documentation which has not been executed yet?
